app/models/unlocalized_words.php

```diff
@@ -0,0 +1,104 @@
+<?php
+namespace Transvision;
+
+// Build arrays for the search form, ignore mozilla_org
+$channel_selector = Utils::getHtmlSelectOptions(
+    $repos_nice_names,
+    $repo,
+    true
+);
+$target_locales_list = Utils::getHtmlSelectOptions(
+    Project::getRepositoryLocales($repo),
+    $locale
+);
+
+$ref_locale = Project::getReferenceLocale($repo);
+
+/*
+    pspell helps getting rid of false positive results by keeping only valid
+    English words. The downside is that it’s filtering out 'jargon' words that
+    can be used in devtools or Mozilla-specific words.
+
+    This requires the following packages to be installed:
+    sudo apt-get install libpspell-dev php5-pspell aspell-en
+*/
+$pspell_link = \pspell_new('en_US', '', '', '', PSPELL_FAST);
+$english_words = $unlocalized_words = [];
+
+// Load reference strings
+$strings_reference = array_map('strtolower', Utils::getRepoStrings(
+    $ref_locale,
+    $repo
+));
+
+// Load locale strings
+$strings_locale = array_map('strtolower', Utils::getRepoStrings($locale, $repo));
+
+/*
+    Go through all strings in $strings_reference, extract valid English words
+    then check if any of them is present in the localized string from
+    $strings_locale
+*/
+foreach ($strings_reference as $string_ref_id => $ref_words) {
+    $english_words = [];
+    $locale_words = $strings_locale[$string_ref_id];
+
+    /*
+        If the string is missing in the locale or has been copy pasted from
+        source (e.g. not translated), skip it
+    */
+    if (! isset($locale_words)
+        || ($ref_words == $locale_words && $locale != $ref_locale)) {
+        continue;
+    }
+
+    /*
+        Remove punctuation characters from the strings then explode them into
+        words
+    */
+    $ref_words    = explode(' ',
+        preg_replace('/(?![.=-])\p{P}/u', '', $ref_words)
```

Comment:
My level of understanding regexps is pretty low, so I think = and - can be removed, but not sure if . is used as a special character, or if it's meant to match with periods. If it's matching periods, we'd need it.